app/src/main/java/org/mozilla/focus/settings/privacy/TrackingProtectionPanel.kt

```diff
@@ -8,45 +8,66 @@ import android.view.View
 import android.widget.FrameLayout
 import androidx.appcompat.content.res.AppCompatResources
 import androidx.core.view.isVisible
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ViewTreeLifecycleOwner
+import androidx.savedstate.SavedStateRegistryOwner
+import androidx.savedstate.setViewTreeSavedStateRegistryOwner
 import com.google.android.material.bottomsheet.BottomSheetBehavior
 import com.google.android.material.bottomsheet.BottomSheetDialog
 import mozilla.components.browser.icons.IconRequest
+import mozilla.components.lib.state.ext.observeAsComposableState
 import mozilla.components.support.ktx.android.view.putCompoundDrawablesRelativeWithIntrinsicBounds
 import mozilla.components.support.ktx.kotlin.tryGetHostFromUrl
 import org.mozilla.focus.R
+import org.mozilla.focus.cookiebannerexception.CookieBannerExceptionItem
+import org.mozilla.focus.cookiebannerexception.CookieBannerExceptionStore
 import org.mozilla.focus.databinding.DialogTrackingProtectionSheetBinding
 import org.mozilla.focus.engine.EngineSharedPreferencesListener.TrackerChanged
 import org.mozilla.focus.ext.components
 import org.mozilla.focus.ext.installedDate
 import org.mozilla.focus.ext.settings
+import org.mozilla.focus.ui.theme.FocusTheme
 
 @SuppressWarnings("LongParameterList")
 class TrackingProtectionPanel(
     context: Context,
+    private val lifecycleOwner: LifecycleOwner,
+    private val cookieBannerExceptionStore: CookieBannerExceptionStore,
     private val tabUrl: String,
     private val blockedTrackersCount: Int,
     private val isTrackingProtectionOn: Boolean,
     private val isConnectionSecure: Boolean,
     private val toggleTrackingProtection: (Boolean) -> Unit,
     private val updateTrackingProtectionPolicy: (String?, Boolean) -> Unit,
     private val showConnectionInfo: () -> Unit,
+    private val showCookieBannerExceptionsDetailsPanel: () -> Unit,
 ) : BottomSheetDialog(context) {
 
     private var binding: DialogTrackingProtectionSheetBinding =
         DialogTrackingProtectionSheetBinding.inflate(layoutInflater, null, false)
 
     init {
+        initWindow()
         setContentView(binding.root)
         expand()
-
         updateTitle()
         updateConnectionState()
         updateTrackingProtection()
         updateTrackersBlocked()
         updateTrackersState()
+        updateCookieBannerException()
         setListeners()
     }
 
+    private fun initWindow() {
+        this.window?.decorView?.let {
+            ViewTreeLifecycleOwner.set(it, lifecycleOwner)
+            it.setViewTreeSavedStateRegistryOwner(
+                lifecycleOwner as SavedStateRegistryOwner,
+            )
+        }
+    }
```

Comment:
Sorry, I'm not that familiar with the approach, could you help me to clarify why we need it ? :)

Reply:
I need this because TrackingProtectionPanel is a BottomSheetDialog that is made with binding XML, and if I don't put it in I will get a  java.lang.IllegalStateException: ViewTreeLifecycleOwner not found from android.widget.FrameLayout.